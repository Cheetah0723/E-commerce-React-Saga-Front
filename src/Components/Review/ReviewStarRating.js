import React from "react";
import StyledRating from "./StyledRating"
import StarIcon from '@material-ui/icons/Star';
import StarBorderIcon from '@material-ui/icons/StarBorder';

export default function ReviewStarRating({ value, readOnly, name, handleClick }) {

    return (
        <StyledRating
            name={name}
            value={value === undefined ? 0 : parseInt(value)}
            readOnly={readOnly}
            precision={0.5}
            icon={<StarIcon fontSize="3rem" />}
            emptyIcon={<StarBorderIcon fontSize="3rem" />}
            onChange={(event, newValue) => {
                if (!readOnly) {
                    handleClick(newValue)
                }
            }}
        />
    )
}