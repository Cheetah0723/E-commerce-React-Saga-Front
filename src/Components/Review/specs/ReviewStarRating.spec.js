import React from 'react'
import ReviewStarRating from "../ReviewStarRating"
import renderer from 'react-test-renderer'

describe.skip('ReviewStarRating', () => {
    it('renders correctly when it is read only', () => {
        const test = (arg) => { console.log(arg) }
        const component = renderer.create(<ReviewStarRating handleClick={test} readOnly={true} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when it is not read only', () => {
        const test = (arg) => { console.log(arg) }
        const component = renderer.create(<ReviewStarRating handleClick={test} readOnly={true} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})