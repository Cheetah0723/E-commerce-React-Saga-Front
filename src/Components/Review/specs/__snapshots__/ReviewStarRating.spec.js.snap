// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewStarRating renders correctly when it is not read only 1`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root MuiRating-readOnly"
  onClick={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`ReviewStarRating renders correctly when it is read only 1`] = `
<span
  aria-label="null Stars"
  className="MuiRating-root MuiRating-readOnly"
  onClick={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
  <span>
    <span
      className="MuiRating-icon MuiRating-iconEmpty"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </span>
  </span>
</span>
`;
